modules/rte_mis_school/src/Access/SchoolDetailEditAccessCheck.php

```diff
@@ -93,6 +106,12 @@ public function access(AccountInterface $account, RouteMatchInterface $routeMatc
               $locationIds[] = $value->tid;
             }
           }
+          // Check if there is an active window for reimbursment claim.
+          // If yes, then allow block admin.
+          $reimbursmentClaimCampaignStatus = $this->rteCoreHelper->isAcademicSessionValid('reimbursement_claim');
+          if ($reimbursmentClaimCampaignStatus && $display == 'school_fee_modify') {
+            return AccessResult::allowed()->setCacheMaxAge(0);
```

Comment:
Don't you think this maxage will have a performance impact while loading the reimbursement claim dashboard

Reply:
It will not impact the performance of the reimbursement claim dashboard, but yes it will impact the performance of the School Fees Modify dashboard for the block admin, every time the block tries to update the fees it will re-evaluate for access rather than considering the data from the cached memory. I'll surely remove this.